test/com/amazon/ionschema/ResourceAuthorityTest.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionschema
+
+import com.amazon.ion.system.IonSystemBuilder
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Assert.fail
+import org.junit.Test
+
+class ResourceAuthorityTest {
+
+    @Test
+    fun unknownSchemaId() {
+        val iss = IonSchemaSystemBuilder.standard().build()
+        val authority = ResourceAuthority("ion-schema-schemas", ResourceAuthority::class.java.classLoader)
+        val iter = authority.iteratorFor(iss, "unknown_schema_id")
+        assertFalse(iter.hasNext())
+        try {
+            iter.next()
+            fail()
+        } catch (e: NoSuchElementException) {
+        }
```

Comment:
This sent me on a distraction reading about `try-with-resources` support in Kotlin (or lack thereof), the first ultimately sad experience I've had with the language. 

There is [`use`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) though.

Reply:
Also it would be great to hop to JUnit5 and use `assertThrows` here.